rules/nette/tests/Rector/Class_/ExtractTemplateClassForPresenterRector/Fixture/some_class.php.inc

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Rector\Nette\Tests\Rector\Class_\ExtractTemplateClassForPresenterRector\Fixture;
+
+use Nette\Application\UI\Presenter;
+
+final class SomePresenter extends Presenter
```

Comment:
what about presenter with more actions?

Reply:
That was my first question when I saw this feature. I haven't used it yet in our projects.

How is that handled in practise?

Reply:
I don't know what is the best practise or recommended way but I use it this way:

```
use Nette\Application\UI\Presenter;

final class SomePresenter extends Presenter
{
    public function renderDefault()
    {
        /** @var SomeDefaultTemplate $template */
        $template = $this->template;
    }

    public function actionCreate()
    {
        /** @var SomeCreateTemplate $template */
        $template = $this->template;
    }

    public function actionEdit(string $id)
    {
        /** @var SomeEditTemplate $template */
        $template = $this->template;
    }
}
```

Reply:
I see. Btw, how does Nette instantiates e.g. `SomeEditTemplate` object for last method? I could not find it

Reply:
I've read somewhere that it should work automatically based on the name of presenter and template (probably only one-action-presenters). But now it seems you have to pass template class name to createTemplate() method and the template with the same type will be created.

So my previous example should be corrected to:

```
use Nette\Application\UI\Presenter;

final class SomePresenter extends Presenter
{
    public function renderDefault()
    {
        /** @var SomeDefaultTemplate $template */
        $template = $this->createTemplate(SomeDefaultTemplate::class);
    }

    public function actionCreate()
    {
        /** @var SomeCreateTemplate $template */
        $template = $this->createTemplate(SomeCreateTemplate::class);
    }

    public function actionEdit(string $id)
    {
        /** @var SomeEditTemplate $template */
        $template = $this->createTemplate(SomeEditTemplate::class);
    }
}
```

Reply:
> I've read somewhere that it should work automatically based on the name of presenter and template (probably only one-action-presenters). 

This works thanks to method Control::formatTemplateClass() but those per-action templates are not implemented. Maybe it is not possible

Reply:
> I see. Btw, how does Nette instantiates e.g. `SomeEditTemplate` object for last method? I could not find it

I've added PR to solve this feature https://github.com/nette/application/pull/277

Reply:
Thank you for looking into this. If you find any code, please share it. Nette documentation does not explain this much